app/feature/feature-payments/src/main/kotlin/com/hedvig/android/feature/payments2/PaymentOverviewDestination.kt

```diff
@@ -0,0 +1,284 @@
+        endSlot = {
+          Row(
+            Modifier.fillMaxWidth(),
+            horizontalArrangement = Arrangement.End,
+            verticalAlignment = Alignment.CenterVertically,
+          ) {
+            Text(
+              text = uiState.upcomingPayment.toString(),
+              textAlign = TextAlign.End,
+            )
+            Spacer(Modifier.width(4.dp))
+            Icon(
+              imageVector = Icons.Hedvig.ChevronRight,
+              contentDescription = null,
+              tint = MaterialTheme.colorScheme.onSurfaceVariant,
+              modifier = Modifier.size(16.dp),
+            )
+          }
+        },
+      )
+      val dateTimeFormatter = rememberHedvigDateTimeFormatter()
+      Text(
+        text = stringResource(
+          id = R.string.GENERAL_DUE,
+          dateTimeFormatter.format(uiState.dueDate.toJavaLocalDate()),
+        ),
+        color = MaterialTheme.colorScheme.onSurfaceVariant,
+      )
+    }
+  }
+}
+
+@Suppress("UnusedReceiverParameter")
+@Composable
+private fun ColumnScope.PaymentDetails(
+  displayName: String,
+  displayValue: String,
+) {
+  Row(
+    modifier = Modifier
+      .fillMaxWidth()
+      .padding(16.dp),
+    horizontalArrangement = Arrangement.spacedBy(8.dp),
+    verticalAlignment = Alignment.CenterVertically,
+  ) {
+    Row(
+      verticalAlignment = Alignment.CenterVertically,
+      modifier = Modifier.weight(1f),
+    ) {
+      Icon(
+        imageVector = Icons.Hedvig.Payments,
+        contentDescription = null,
+        modifier = Modifier.size(24.dp),
+      )
+      Spacer(Modifier.width(16.dp))
+      Text(displayName)
+    }
+    Spacer(Modifier.width(8.dp))
+    Text(
+      text = displayValue,
+      color = MaterialTheme.colorScheme.onSurfaceVariant,
+    )
+  }
+}
+
+@Composable
+fun Discounts(modifier: Modifier = Modifier) {
+  Row(
+    modifier = modifier.fillMaxWidth(),
+    verticalAlignment = Alignment.CenterVertically,
+  ) {
+    Icon(
+      imageVector = Icons.Hedvig.Campaign,
+      contentDescription = null,
+      tint = MaterialTheme.colorScheme.typeElement,
+      modifier = Modifier.size(24.dp),
+    )
+    Spacer(Modifier.width(16.dp))
+    Text(stringResource(id = R.string.PAYMENTS_DISCOUNTS_SECTION_TITLE))
+  }
+}
+
+
+@Composable
+fun PaymentHistory(modifier: Modifier = Modifier) {
+  Row(
+    modifier = modifier.fillMaxWidth(),
+    verticalAlignment = Alignment.CenterVertically,
+  ) {
+    Icon(
+      imageVector = Icons.Hedvig.Waiting,
+      contentDescription = null,
+      modifier = Modifier.size(24.dp),
+    )
+    Spacer(Modifier.width(16.dp))
+    Text(stringResource(id = R.string.PAYMENTS_PAYMENT_HISTORY_BUTTON_LABEL))
+  }
+}
+
+@Composable
+@HedvigPreview
+private fun PreviewPaymentScreen(
+  @PreviewParameter(DoubleBooleanCollectionPreviewParameterProvider::class) input: Pair<Boolean, Boolean>,
+) {
+  val hasPaymentConnected = input.first
+  val hasCampaigns = input.second
+  HedvigTheme {
+    Surface {
```

Comment:
```suggestion
    Surface(color = MaterialTheme.colorScheme.background) {
```
To get a better representation of how this will actually look like in the app itself, since the background has background color, not `surface` which the `Surface` composable defaults to.

Can copy https://github.com/HedvigInsurance/android/pull/1677#discussion_r1259903717 this to just write `prev` and get the skeleton autocompleted for you to not have to think about this.